distributed/active_memory_manager.py

```diff
@@ -0,0 +1,429 @@
+import asyncio
+import heapq
+import math
+from collections import defaultdict
+from collections.abc import Generator, Iterable, Iterator
+from typing import Optional
+
+import dask
+
+from .scheduler import SchedulerPlugin, SchedulerState, TaskState, WorkerState
+
+
+class ActiveMemoryManager(SchedulerPlugin):
+    scheduler: SchedulerState
+    plugins: "set[ActiveMemoryManagerPlugin]"
+    interval: float
```

Comment:
In the future I expect that different policies will want to operate on different timelines.  For example policies that are more expensive to run might run much less frequently.

Reply:
This sounds like a desirable feature; however is it ok to leave it for the future?

Reply:
Sounds good to me